### DISCUSSION TITLE
what's the problem here?? ( I attached link)

### DISCUSSION THREAD

**Discussion by NSH531** [dotnet/runtime#73609] [Q&A] [answered]
https://github.com/PowerShell/PowerShell/discussions/17861

**Comment by danmoseley** [comment, marked as answer]
Hello. This is not the right repo for powershell questions. It is the right repo if you trace it to an issue in the underlying libraries.

**Reply by NSH531** [reply]
It's complicated because have a code that runs powershell script:
```
powershell.AddScript(G2);
                results = powershell.Invoke();
                var tj = new List<string>();
                IAsyncResult asyncResult;
                PSDataCollection<string> a2 = new PSDataCollection<string>();
                var a1 = a2;
                a1.Add(string.Format("{0}{3}{1}{3}{2}", SRaUTH, g, G2, "\n"));
                Task<PSDataCollection<PSObject>> T = powershell.InvokeAsync<string, string>(a1, a2);
                for (int Ag = 0; Ag < T.Result.ToArray().Length; Ag++)
                {
                    for (long jh = 0; jh < T.Result[Ag].Properties.Count(); jh++)
                    {
                        label3.Text = label3.Text + "\n\n" + T.Result[Ag].Properties.ToArray()[jh];

                    }
                }
```

**Reply by danmoseley** [reply]
Have you simplified the code as far as you possibly can so you have the minimum code that repros the issue?

**Reply by Clockwork-Muse** [reply]
... you also haven't specified the results that you are getting, versus the results you expect to be getting.  Because of that, and the fact that your code isn't a [complete repro](https://github.com/dotnet/runtime/blob/main/CONTRIBUTING.md#writing-a-good-bug-report), we don't even know whether there's anything actually wrong with your code.

**Comment by NSH531** [comment]
I've attached the relevant code:

```

 Task<PSDataCollection<PSObject>> T = powershell.InvokeAsync<string,
string>(a1, a2);
                for (int Ag = 0; Ag < T.Result.ToArray().Length; Ag++)
                {
                    for (long jh = 0; jh < T.Result[Ag].Properties.Count();
jh++)
                    {
                        label3.Text = label3.Text + "\n\n" +
T.Result[Ag].Properties.ToArray()[jh];

                    }
                }
            }

```



‫בתאריך יום ג׳, 9 באוג׳ 2022 ב-10:34 מאת ‪Dan Moseley‬‏ <‪
***@***.***‬‏>:‬

> Have you simplified the code as far as you possibly can so you have the
> minimum code that repros the issue?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/runtime/discussions/73609#discussioncomment-3355520>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABNBJWKWTFHO3KLSLZ4X2UTVYICYXANCNFSM557SRQEQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<https://netanel.ml>

**Reply by NSH531** [reply]
```
Task<PSDataCollection<PSObject>> T = powershell.InvokeAsync<string,
string>(a1, a2);
                for (int Ag = 0; Ag < T.Result.ToArray().Length; Ag++)
                {
                    for (long jh = 0; jh < T.Result[Ag].Properties.Count();
jh++)
                    {
                        label3.Text = label3.Text + "\n\n" +
T.Result[Ag].Properties.ToArray()[jh];

                    }
                }
            }

```
